lib/types/file.js

```diff
@@ -0,0 +1,104 @@
+export default class File {
+  /* eslint-disable max-params */
+  constructor(
+    checksum,
+    compression,
+    size,
+    writerId,
+    userId,
+    recordType,
+    plain = {},
+    fileUrl = null,
+    fileName = null,
+    recordId = null,
+    created = null,
+    lastModified = null,
+    version = null
+  ) {
+    // Empty dict for data.
+    this.data = {}
+    // Base64 encoded MD5 Checksum of the Encrypted File, including E3DB Header information
+    this.checksum = checksum
+    // The type of compression the file is using, before encryption. Reserved for future use.
+    this.compression = compression
+    // Size of the encrypted file, in bytes, including E3DB Header information
+    this.size = size
+    this.writerId = writerId
+    this.userId = userId
+    this.recordType = recordType
+    // Optional. Signed url used for PUT/GET to storage server
+    this.fileUrl = fileUrl
+    // Optional. Name of the file stored on the server. File name consists of UUID + timestamp. Returned by the server after the file has been uploaded.
+    this.fileName = fileName
+    // Optional. ID of the Record in a UUID format. Returned by the server after the file has been uploaded.
+    this.recordId = recordId
+    // Optional. Created timestamp of the file, returned by the server after the file has been uploaded.
+    this.created = created
+    // Optional. Last Modified timestamp of the file, returned by the server after the file has been uploaded.
+    this.lastModified = lastModified
+    // Optional. The version of the file based on File structure and cryptographic methods.
+    this.version = version
+    // Optional. Plaintext metadata attached to the File.
+    this.plain = plain
+  }
+  /* eslint-enable maxparams */
+
+  removeEmpty(obj) {
+    /*
```

Comment:
Probably should remove or adjust to JS doc these Python comments.

Reply:
> Worth noting now that I have read the PR more closely -- this is a lot of e3db requests, so testing either needs to be heavily mocked, or needs to be integration level. Since this does not have an actual crypto library attached, integration testing is a lot harder. It may be best to have a good suite around the consuming libraries and make sure the code is operational there in a full integration test suite with crypto attached.

I think we should go for integration tests.  The browser crypto needs window.crypto methods.  Jest doesn't have those.  Mocking them is a waste of time.  I think the big value test will be able to tell if something encrypts and decrypts.  So I think the way things are built now, we should hop up to the level of integration tests.